app/pages/ProjectsPage.tsx

```diff
@@ -77,11 +69,8 @@ export function Component() {
         onActivate: () => {
           // the edit view has its own loader, but we can make the modal open
           // instantaneously by preloading the fetch result
-          apiQueryClient.setQueryData(
-            'projectView',
-            { path: { project: project.name } },
-            project
-          )
+          const queryOptions = apiq('projectView', { path: { project: project.name } })
+          queryClient.setQueryData(queryOptions.queryKey, project)
```

Comment:
I was surprised to discover this does actually enforce the type of `project` matches the endpoint. 

<img width="708" alt="image" src="https://github.com/user-attachments/assets/0f22d451-dcf7-48cb-a525-8826012a6198">

`setQueryData` is inferring it from the type of the options object.

<img width="821" alt="image" src="https://github.com/user-attachments/assets/6f4c5249-8127-4bae-bf6a-9680b86a1532">

<img width="730" alt="image" src="https://github.com/user-attachments/assets/b1a5ab65-05a3-4689-9704-6967291bd320">